app/controllers/agents_controller.rb

```diff
@@ -162,24 +162,23 @@ def new
     agents = current_user.agents
 
     if id = params[:id]
```

Comment:
I agree. I was trying to make this as minimal of a change as possible so I could elicit feedback first. I'm happy to do some refactoring of the agent controller as well in this PR or can do that after this one is accepted or not.